hibernate-core/src/test/java/org/hibernate/orm/test/hql/ASTParserLoadingTest.java

```diff
@@ -3665,15 +3674,17 @@ public void testEJBQLFunctions() throws Exception {
 		hql = "select length(a.description) from Animal a";
 		session.createQuery(hql).list();
 
-		//note: postgres and db2 don't have a 3-arg form, it gets transformed to 2-args
-		hql = "from Animal a where locate('abc', a.description, 2) = 2";
-		session.createQuery(hql).list();
+		if ( !( getDialect() instanceof InformixDialect && getDialect().getVersion().isBefore( 12 ) ) ) {
```

Comment:
```suggestion
		// Informix before version 11.70 didn't support finding the index of substrings
		if ( !( getDialect() instanceof InformixDialect && getDialect().getVersion().isBefore( 11, 70 ) ) ) {
```

Reply:
I'll add a comment and return the test for version 12.